src/stale-prs.ts

```diff
@@ -154,8 +154,10 @@ export class StalePrFinder {
         console.log('        Stale since:        ', stale.since.toISOString());
         console.log('        Warnings:           ', Object.fromEntries(warnings)); // Prints nicer
 
+        const memberReviewed = await this.memberReviewed(pull.number);
+
         const warning = warnings.get('STALE PR');
-        if (!warning || warning < stale.since) {
+        if (!warning || warning < stale.since || !memberReviewed) {
```

Comment:
Is this correct? 

This is saying:

1. If we don't have a warning yet
2. OR the last warning we gave is from before the start of the current staleness period
3. OR the PR hasn't been reviewed yet

THEN we should add a warning and start the timer for closing.

For fresh PRs, the 3rd condition will be automatically true, because at the start nobody will have looked at it yet.

So this would add a staleness warning immediately on every new PR!

-----

I think what you probably want is to have `memberReviewed` factor into the value of `stale`: a PR should *not* be considered stale if it hasn't been reviewed yet.

Have a look at line 115, and think about how your change would interact with the conditions that are laid out there -- then make the changes over there.

Reply:
Oh good catch, I was thinking this would only be run if the PR is already stale but the `stale` variable includes any build failures and merge conflicts as well. I'll update it with your suggestion.